src/test/kotlin/org/elm/ide/inspections/inference/TypeInferenceInspectionTest1.kt

```diff
@@ -290,6 +290,15 @@ main : ()
 main = foo .x
 """)
 
+    fun `test field accessor as argument to type var in pipeline`() = checkByText("""
+foo : (a -> ()) -> a -> a
+foo _ a = a
+
+main : ()
+main =
+    <error descr="Type mismatch.Required: ()Found: { a | f2 : (), f1 : () }">{ f1 = (), f2 = () } |> foo .f2</error>
```

Comment:
This isn't entirely correct: the actual type should be `{ f2 : (), f1 : () }` i.e. not a subset. But this won't cause false positives, and will only cause false negatives in very contrived code. It doesn't seem worth fixing right now.